code/modules/mob/living/simple_animal/friendly/spiderbot.dm

```diff
@@ -1,53 +1,53 @@
 /mob/living/simple_animal/spiderbot
-
-	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
-	minbodytemp = 0
-	maxbodytemp = 500
-
-	var/obj/item/device/radio/borg/radio = null
-	var/mob/living/silicon/ai/connected_ai = null
-	var/obj/item/weapon/stock_parts/cell/cell = null
-	var/obj/machinery/camera/camera = null
-	var/obj/item/device/mmi/mmi = null
-	var/list/req_access = list(access_robotics) //Access needed to pop out the brain.
-
 	name = "Spider-bot"
-	desc = "A skittering robotic friend!"
+	desc = "A skittering robotic friend!" //More like ultimate shitter
 	icon = 'icons/mob/robots.dmi'
 	icon_state = "spiderbot-chassis"
 	icon_living = "spiderbot-chassis"
 	icon_dead = "spiderbot-smashed"
 	wander = 0
 	universal_speak = 1
-	health = 10
-	maxHealth = 10
+	health = 40
+	maxHealth = 40
 
+	melee_damage_lower = 2
+	melee_damage_upper = 2
+	melee_damage_type = BURN
 	attacktext = "shocks"
-	attacktext = "shocks"
-	melee_damage_lower = 1
-	melee_damage_upper = 3
+	attack_sound = "sparks"
 
 	response_help  = "pets"
 	response_disarm = "shoos"
 	response_harm   = "stomps on"
-
-	var/obj/item/held_item = null //Storage for single item they can hold.
-	var/emagged = 0               //IS WE EXPLODEN?
-	var/syndie = 0                //IS WE SYNDICAT? (currently unused)
-	speed = -1                    //Spiderbots gotta go fast.
-	//pass_flags = PASSTABLE      //Maybe griefy?
+	speed = 0
 	mob_size = MOB_SIZE_SMALL
 	speak_emote = list("beeps","clicks","chirps")
+
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	minbodytemp = 0
+	maxbodytemp = 500
+
 	can_hide = 1
 	ventcrawler = 2
 	loot = list(/obj/effect/decal/cleanable/blood/gibs/robot)
 	del_on_death = 1
 
-/mob/living/simple_animal/spiderbot/attackby(var/obj/item/O as obj, var/mob/user as mob, params)
+	var/emagged = 0               //is it getting ready to explode?
+	var/obj/item/device/mmi/mmi = null
+	var/emagged_master = null //for administrative purposes, to see who emagged the spiderbot; also for a holder for if someone emags an empty frame first then inserts an MMI.
+
+/mob/living/simple_animal/spiderbot/Destroy()
+	if(emagged)
+		QDEL_NULL(mmi)
```

Comment:
Should this ghost the mind mob before being deleted, or is that handled already in the Destroy() for the MMI/whatever?